### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by koep**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/33656)

### CODE REVIEW

File: staging/src/k8s.io/client-go/1.5/_vendor/github.com/emicklei/go-restful/CHANGES.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 Change history of go-restful
 =
 2016-02-14
-- take the qualify factor of the Accept header mediatype into account when deciding the contentype of the response
```
Comment (thockin) [inline staging/src/k8s.io/client-go/1.5/_vendor/github.com/emicklei/go-restful/CHANGES.md]: We don't edit things in vendor directories.

**Review by thockin — CHANGES_REQUESTED** [review]
LG overall.  1 file to revert, and needs rebase

**Comment by koep** [comment]
Thanks @thockin. I think I fixed everything now.

**Comment by koep** [comment]
Btw do you care about this kind of changes?
Or is it more of a burden for you to review them?

**Comment by thockin** [comment]
ok to test

**Review by thockin — APPROVED** [review]

**Comment by thockin** [comment]
Thanks for the PR.  Fixing typos is important, and I try to share the overall code-review load :)

**Comment by thockin** [comment]
Can you run `make verify` against this and fix the issues?  Thanks.

**Comment by koep** [comment]
I fixed some errors with the ```hack/.linted_packages``` file. 
Other than that I got three (probably unrelated) errors from 
```
hack/make-rules/../../hack/verify-golint.sh
hack/make-rules/../../hack/verify-govet.sh
hack/make-rules/../../hack/verify-munge-docs.sh
```
For instance
```
Errors from golint:
pkg/util/metrics/util.go:37:21: should omit type map[string]prometheus.Gauge from declaration of var rateLimiterMetrics; it will be inferred from the right-hand side
```
and
```
pkg/kubelet/kubelet_test.go:708: possible formatting directive in Error call
pkg/kubelet/kubelet_test.go:716: possible formatting directive in Error call
pkg/kubelet/kubelet_test.go:722: possible formatting directive in Error call
exit status 1
pkg/security/apparmor/validate_test.go:76: possible formatting directive in Error call
pkg/security/apparmor/validate_test.go:130: possible formatting directive in Error call
pkg/security/apparmor/validate_test.go:157: possible formatting directive in Error call
exit status 1
pkg/security/podsecuritypolicy/sysctl/mustmatchpatterns_test.go:92: k8s.io/kubernetes/pkg/api.Sysctl composite literal uses unkeyed fields
exit status 1
plugin/pkg/scheduler/algorithm/priorities/priorities_test.go:385: k8s.io/kubernetes/plugin/pkg/scheduler/api.HostPriority composite literal uses unkeyed fields
plugin/pkg/scheduler/algorithm/priorities/priorities_test.go:385: k8s.io/kubernetes/plugin/pkg/scheduler/api.HostPriority composite literal uses unkeyed fields
plugin/pkg/scheduler/algorithm/priorities/priorities_test.go:402: k8s.io/kubernetes/plugin/pkg/scheduler/api.HostPriority composite literal uses unkeyed fields
plugin/pkg/scheduler/algorithm/priorities/priorities_test.go:402: k8s.io/kubernetes/plugin/pkg/scheduler/api.HostPriority composite literal uses unkeyed fields
plugin/pkg/scheduler/algorithm/priorities/priorities_test.go:419: k8s.io/kubernetes/plugin/pkg/scheduler/api.HostPriority composite literal uses unkeyed fields
plugin/pkg/scheduler/algorithm/priorities/priorities_test.go:419: k8s.io/kubernetes/plugin/pkg/scheduler/api.HostPriority composite literal uses unkeyed fields
plugin/pkg/scheduler/algorithm/priorities/priorities_test.go:442: k8s.io/kubernetes/plugin/pkg/scheduler/api.HostPriority composite literal uses unkeyed fields
plugin/pkg/scheduler/algorithm/priorities/priorities_test.go:442: k8s.io/kubernetes/plugin/pkg/scheduler/api.HostPriority composite literal uses unkeyed fields
plugin/pkg/scheduler/algorithm/priorities/priorities_test.go:1145: k8s.io/kubernetes/plugin/pkg/scheduler/api.HostPriority composite literal uses unkeyed fields
plugin/pkg/scheduler/algorithm/priorities/priorities_test.go:1145: k8s.io/kubernetes/plugin/pkg/scheduler/api.HostPriority composite literal uses unkeyed fields
plugin/pkg/scheduler/algorithm/priorities/priorities_test.go:1145: k8s.io/kubernetes/plugin/pkg/scheduler/api.HostPriority composite literal uses unkeyed fields
exit status 1
make: *** [Makefile:231: vet] Error 1
```
and also
```
make: Entering directory '/home/cko/.go/src/k8s.io/kubernetes'
make[1]: Entering directory '/home/cko/.go/src/k8s.io/kubernetes'
make[1]: Leaving directory '/home/cko/.go/src/k8s.io/kubernetes'
+++ [1002 00:42:11] Generating bindata:
    /home/cko/.go/src/k8s.io/kubernetes/test/e2e/framework/gobindata_util.go
+++ [1002 00:42:11] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
+++ [1002 00:42:11] Building go targets for linux/amd64:
    cmd/mungedocs
make: Leaving directory '/home/cko/.go/src/k8s.io/kubernetes'
/home/cko/.go/src/k8s.io/kubernetes/hack/lib/util.sh: line 283: mdfiles[@]: unbound variable
!!! Error in hack/make-rules/../../hack/verify-munge-docs.sh:73
  'needsanalytics=($(kube::util::gen-analytics "${KUBE_ROOT}" 1))' exited with status 1
Call stack:
  1: hack/make-rules/../../hack/verify-munge-docs.sh:73 main(...)
Exiting with status 1
```

**Comment by koep** [comment]
I also ran ```make verify``` against master and it also fails.

File: build/README.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ Building Kubernetes is easy if you take advantage of the containerized build env
 ## Requirements
 
 1. Docker, using one of the two following configurations:
-  1. **Mac OS X** You can either use docker-machine or boot2docker. See installation instructions [here](https://docs.docker.com/installation/mac/).
```
Comment (ultimateboy) [inline build/README.md]: Double spaces at the end of a line convert to a linebreak in markdown. These should be added back to give the "note" more visibility.

File: hack/.linted_packages [hunk 3]
```diff
@@ -6,18 +6,18 @@ cmd/genkubedocs
 cmd/genman
 cmd/genswaggertypedocs
 cmd/integration
+cmd/kubeadm
+cmd/kubeadm
```
Comment (ultimateboy) [inline hack/.linted_packages]: duplicate

**Review by ultimateboy — COMMENTED** [review]
Reviewed all of the changes - primarily spelling, capitalization, and whitespace removals. All but a single change seem correct to me where a double-space at the end of a line was removed. That double space was being converted to a linebreak and should probably be preserved to maintain readability. Otherwise, despite the size of this PR, all the changes seem appropriate.

File: build/README.md (hunk 2 above)
Comment (koep) [inline build/README.md]: I don't see any difference

https://github.com/kubernetes/kubernetes/blob/master/build/README.md
https://github.com/koep/kubernetes/blob/7715d74af327fa40aaffa1fa640649c0a38a4b49/build/README.md

**Comment by thockin** [comment]
@k8s-bot verify test this

**Comment by thockin** [comment]
@k8s-bot node e2e test this

**Comment by thockin** [comment]
It still says verify-golint.sh is failing

**Comment by koep** [comment]
I really don't get why. It works on my box.
```
$ hack/make-rules/../../hack/verify-golint.sh
examples/examples_test.go is in package examples_test, not examples
pkg/api/meta_test.go is in package api_test, not api
pkg/api/meta/help_test.go is in package meta_test, not meta
pkg/api/resource/quantity_example_test.go is in package resource_test, not resource
pkg/api/v1/types.generated.go is in package v1, not v1_test
pkg/apis/abac/v0/conversion_test.go is in package v0_test, not v0
pkg/apis/batch/v1/defaults_test.go is in package v1_test, not v1
pkg/apis/batch/v2alpha1/defaults_test.go is in package v2alpha1_test, not v2alpha1
pkg/apis/extensions/v1beta1/conversion_test.go is in package v1beta1_test, not v1beta1
pkg/client/testing/core/fake_test.go is in package core_test, not core
pkg/client/unversioned/scheduledjobs_test.go is in package unversioned_test, not unversioned
pkg/client/unversioned/auth/clientauth_test.go is in package auth_test, not auth
pkg/conversion/queryparams/convert_test.go is in package queryparams_test, not queryparams
pkg/kubelet/container/runtime_cache_test.go is in package container_test, not container
pkg/kubelet/envvars/doc.go is in package envvars, not envvars_test
pkg/proxy/config/doc.go is in package config, not config_test
pkg/runtime/extension_test.go is in package runtime_test, not runtime
pkg/runtime/serializer/json/json_test.go is in package json_test, not json
pkg/runtime/serializer/protobuf/protobuf_test.go is in package protobuf_test, not protobuf
pkg/serviceaccount/jwt_test.go is in package serviceaccount_test, not serviceaccount
pkg/storage/selection_predicate_test.go is in package storage, not storage_test
pkg/util/io/io_test.go is in package io_test, not io
pkg/util/workqueue/rate_limitting_queue_test.go is in package workqueue, not workqueue_test
pkg/volume/metrics_du_test.go is in package volume_test, not volume
pkg/watch/versioned/encoder_test.go is in package versioned_test, not versioned
Congratulations!  All Go source files have been linted.

Success! All packages that should pass lint are listed in the linted file.
```

Do you want me to open a new, clean PR with just the changes you approved? Maybe something went wrong while rebasing.

**Comment by koep** [comment]
It looks like Jenkins is ignoring my updated ```hack/.linted_packages``` file.
Like I said: ```make verify``` returns no errors on my machine.

**Comment by thockin** [comment]
please rebase and I will check it out and see what is wrong

**Comment by thockin** [comment]
I get

```
hack/.linted_packages is not in alphabetical order. Please sort it:

  LANG=C sort -o hack/.linted_packages hack/.linted_packages
```

**Comment by koep** [comment]
Here is what I did:
```
git fetch upstream master
```
```
git rebase upstream/master
```
```
make verify
```
```
$ echo $?
0
```
```
git push origin typos -f
```

**Comment by thockin** [comment]
if you run `LANG=C sort -o hack/.linted_packages hack/.linted_packages`
does it change the file?

**Comment by koep** [comment]
Nope, not anymore. But it did in the past (see changed files in this PR).

```
cko ~> arch ~/go/src/k8s.io/kubernetes on typos 
$ LANG=C sort -o hack/.linted_packages hack/.linted_packages
```
```
cko ~> arch ~/go/src/k8s.io/kubernetes on typos 
$ git status
On branch typos
Your branch is up-to-date with 'origin/typos'.
nothing to commit, working tree clean
```

**Comment by thockin** [comment]
What OS are you on?

Try this:

```
$ echo -e "test/images/port-forward-tester\ntest/images/porter" | sort
test/images/porter
test/images/port-forward-tester

$ echo -e "test/images/port-forward-tester\ntest/images/porter" | LANG=C sort
test/images/port-forward-tester
test/images/porter

$ sort --version
sort (GNU coreutils) 8.21
Copyright (C) 2013 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by Mike Haertel and Paul Eggert.

```

@eparis @david-mcmahon

**Comment by eparis** [comment]
To be fair, I see some thing (including `verify-golint.sh`) are improperly doing what @thockin did. It is using `LANG=C` but should be using `LC_ALL=C`. As I believe LC_ALL will overwrite the LANG!

**Comment by koep** [comment]
I'm running Arch Linux.
```bash
$ uname -a
Linux WS999DSL042346 4.7.6-1-ARCH #1 SMP PREEMPT Fri Sep 30 19:28:42 CEST 2016 x86_64 GNU/Linux
```

Also:
```bash
cko ~> host ~/go/src/k8s.io/kubernetes on typos 
$ echo -e "test/images/port-forward-tester\ntest/images/porter" | sort
test/images/porter
test/images/port-forward-tester

cko ~> host ~/go/src/k8s.io/kubernetes on typos 
$ echo -e "test/images/port-forward-tester\ntest/images/porter" | LANG=C sort
test/images/porter
test/images/port-forward-tester

cko ~> host ~/go/src/k8s.io/kubernetes on typos 
$ sort --version
sort (GNU coreutils) 8.25
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by Mike Haertel and Paul Eggert.
```

**Comment by eparis** [comment]
@koep can you try one more? `echo -e "test/images/port-forward-tester\ntest/images/porter" | LC_ALL=C sort`

**Comment by koep** [comment]
Sure.
```bash
$ echo -e "test/images/port-forward-tester\ntest/images/porter" | LC_ALL=C sort
test/images/port-forward-tester
test/images/porter
```

**Comment by eparis** [comment]
boom, I'll open a separate PR to fix that....

**Comment by eparis** [comment]
https://github.com/kubernetes/kubernetes/pull/35120

**Comment by thockin** [comment]
That is approved.  Once it merges, please run the update-all one more time
:)

On Oct 19, 2016 6:51 AM, "Eric Paris" <notifications@github.com> wrote:

#35120 <https://github.com/kubernetes/kubernetes/pull/35120>

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/33656#issuecomment-254818388>,
or mute the thread

**Comment by koep** [comment]
ping @thockin

**Pull request merged**